src/credentials/GroupDataProviderImpl.h

```diff
@@ -19,6 +19,32 @@
 #include <credentials/GroupDataProvider.h>
 #include <lib/core/CHIPPersistentStorageDelegate.h>
 #include <lib/support/Pool.h>
+#include <lib/support/SafeInt.h>
+#include <platform/CHIPDeviceLayer.h>
+#include <platform/KeyValueStoreManager.h>
+
+namespace chip {
+
+class GroupStorageDelegate : public chip::PersistentStorageDelegate
+{
+public:
+    CHIP_ERROR
+    SyncGetKeyValue(const char * key, void * buffer, uint16_t & size) override
+    {
+        return chip::DeviceLayer::PersistedStorage::KeyValueStoreMgr().Get(key, buffer, size);
+    }
+
+    CHIP_ERROR SyncSetKeyValue(const char * key, const void * value, uint16_t size) override
+    {
+        return chip::DeviceLayer::PersistedStorage::KeyValueStoreMgr().Put(key, value, size);
+    }
+
+    CHIP_ERROR SyncDeleteKeyValue(const char * key) override
+    {
+        return chip::DeviceLayer::PersistedStorage::KeyValueStoreMgr().Delete(key);
+    }
+};
+} // namespace chip
```

Comment:
This is purely a forwarding. It should instead be that GroupDataProviderImpl gets passed a PersistentStorageDelegate. Otherwise it bypasses common code that may be implemented for a platform in PersistentStorage delegate and forces changes to this code through forking to do something other than `chip::DeviceLayer::PersistedStorage::KeyValueStoreMgr()`.

Please inject a `PersistentStorageDelegate *` at construction and/or in a custom init call for the `GroupDataProviderImpl` version of the `GroupDataProvider` interface